lib/OnyxCache.ts

```diff
@@ -0,0 +1,128 @@
+import {deepEqual} from 'fast-equals';
+import utils from './utils';
+
+type StorageMap = Record<string, unknown>;
+
+/**
+ * In memory cache providing data by reference
+ * Encapsulates Onyx cache related functionality
+ */
+class OnyxCache {
+    private storageKeys: Set<string>;
+
+    private recentKeys: Set<string>;
+
+    private storageMap: StorageMap;
+
+    private pendingPromises: Map<string, Promise<unknown>>;
+
+    private maxRecentKeysSize = 0;
+
+    constructor() {
+        this.storageKeys = new Set();
+        this.recentKeys = new Set();
+        this.storageMap = {};
+        this.pendingPromises = new Map();
```

Comment:
Shouldn't we keep the bindAll thing?

```ts
        // bind all public methods to prevent problems with `this`
        _.bindAll(
            this,
            'getAllKeys',
            'getValue',
            'hasCacheForKey',
            'addKey',
            'set',
            'drop',
            'merge',
            'hasPendingTask',
            'getTaskPromise',
            'captureTask',
            'removeLeastRecentlyUsedKeys',
            'setRecentKeysLimit',
        );
```

Reply:
Looking at the usage I think it is unnecessary, but for sake of mind I can leave ii 👍